bundles/org.openhab.binding.webexteams/src/main/java/org/openhab/binding/webexteams/internal/WebexAuthService.java

```diff
@@ -0,0 +1,184 @@
+    protected void deactivate(ComponentContext componentContext) {
+        logger.debug("Deactivating WebexAuthService");
+        httpService.unregister(WEBEX_ALIAS);
+        httpService.unregister(WEBEX_ALIAS + WEBEX_RES_ALIAS);
+    }
+
+    /**
+     * Creates a new {@link WebexAuthServlet}.
+     *
+     * @return the newly created servlet
+     * @throws IOException thrown when an HTML template could not be read
+     */
+    private HttpServlet createServlet() throws IOException {
+        return new WebexAuthServlet(this, readTemplate(TEMPLATE_INDEX), readTemplate(TEMPLATE_ACCOUNT));
+    }
+
+    /**
+     * Reads a template from file and returns the content as String.
+     *
+     * @param templateName name of the template file to read
+     * @return The content of the template file
+     * @throws IOException thrown when an HTML template could not be read
+     */
+    private String readTemplate(String templateName) throws IOException {
+        final URL index = bundleContext.getBundle().getEntry(templateName);
+
+        if (index == null) {
+            throw new FileNotFoundException(
+                    String.format("Cannot find '{}' - failed to initialize Webex servlet", templateName));
+        } else {
+            try (InputStream inputStream = index.openStream()) {
+                return new String(inputStream.readAllBytes(), StandardCharsets.UTF_8);
+            }
+        }
+    }
+
+    /**
+     * Call with Spotify redirect uri returned State and Code values to get the refresh and access tokens and persist
+     * these values
+     *
+     * @param servletBaseURL the servlet base, which will be the Spotify redirect url
+     * @param state The Spotify returned state value
+     * @param code The Spotify returned code value
+     * @return returns the name of the Spotify user that is authorized
+     * @throws WebexTeamsException if no handler was found for the state
+     */
+    public String authorize(String servletBaseURL, String state, String code) throws WebexTeamsException {
+        logger.debug("Authorizing for state: {}, code: {}", state, code);
+
+        final WebexTeamsHandler listener = getWebexTeamsHandler(state);
+
+        if (listener == null) {
+            logger.debug(
+                    "Webex redirected with state '{}' but no matching account was found. Possible account has been removed.",
+                    state);
+            throw new WebexTeamsException(ERROR_UKNOWN_BRIDGE);
+        } else {
+            return listener.authorize(servletBaseURL, code);
+        }
+    }
+
+    /**
+     * @param listener Adds the given handler
+     */
+    public void addWebexTeamsHandler(WebexTeamsHandler listener) {
+        if (!handlers.contains(listener)) {
+            handlers.add(listener);
+        }
+    }
+
+    /**
+     * @param handler Removes the given handler
+     */
+    public void removeWebexTeamsHandler(WebexTeamsHandler handler) {
+        handlers.remove(handler);
+    }
```

Comment:
As `handlers` is not thread safe, maybe you should add "synchronized" on methods ?

Reply:
Or change for a thread-safe list.

Reply:
Change to thread safe list